src/plugins/kibana/server/lib/case_conversion.js

```diff
@@ -0,0 +1,15 @@
+const _ = require('lodash');
+
+module.exports = {
+  convertToSnakeCase: function (object) {
+    return _.mapKeys(object, (value, key) => {
+      return _.snakeCase(key);
+    });
+  },
+
+  convertToCamelCase: function (object) {
```

Comment:
Originally my intent was to follow lodash's convention (shallow by default), but it's true that lodash is inconsistent. `_.isEqual` is also deep by default. That inconsistency and the confusion it just caused here is probably reason enough to be more verbose in our own method names, so I updated them to `keysToSnakeCaseShallow` and `keysToCamelCaseShallow`.